Signal/src/ViewControllers/MessagesViewController.m

```diff
@@ -3074,17 +3072,28 @@ - (void)startRecordingVoiceMemo
 
     if (![self.audioRecorder prepareToRecord]) {
         DDLogError(@"%@ audioRecorder couldn't prepareToRecord.", self.tag);
-        [self resetRecordingVoiceMemo];
+        [self cancelVoiceMemo];
         OWSAssert(0);
         return;
     }
 
     if (![self.audioRecorder record]) {
         DDLogError(@"%@ audioRecorder couldn't record.", self.tag);
-        [self resetRecordingVoiceMemo];
+        [self cancelVoiceMemo];
         OWSAssert(0);
         return;
     }
+
+    if (session.recordPermission != AVAudioSessionRecordPermissionGranted) {
+        DDLogError(@"%@ we do not have recording permission.", self.tag);
+        [self cancelVoiceMemo];
+        [ViewControllerUtils
```

Comment:
Good idea.

I've made a new class where we can hang reusable alerts.

 https://github.com/WhisperSystems/Signal-iOS/pull/2070/commits/11fa9d84f7320d2b55c32f1acded1f766ba6a12a